### PR TITLE
Add olive-aml integration doc

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

- Add Azure Arc doc
- Add olive-aml integration doc

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: docs/source/features/azureml_integration.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Azure ML integration
+
+This documents outlines the integrations between Olive and Azure Machine Learning. Discover how to optimize your Azure Machine Learning assets within Olive.
+
```
Comment (devang-ml) [inline docs/source/features/azureml_integration.md:4]: .... how to _use_ your Azure Machine Learning assets ... ?

File: docs/source/features/azureml_integration.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+"model_path": {
+    "type": "azureml_model",
+    "config": {
+        "name": "<model_name>",
+        "version": "<model_version>"
+    }
+ }
+```
+Olive will automatically download the model and run the workflow with this model as input model.
+
```
Comment (devang-ml) [inline docs/source/features/azureml_integration.md:18]: will it run the workflow locally? Please clarify.

File: docs/source/features/azureml_integration.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+### Using model stored in AzureML datastore
+You can specify your model path from an AzureML datastore as:
+```
```
Comment (devang-ml) [inline docs/source/features/azureml_integration.md:21]: Do you mean models from the curated catalogue?

Comment (xiaoyu-work) [reply docs/source/features/azureml_integration.md:21]: No, this is from user datastore. They can store their model in their datastore.

File: docs/source/features/azureml_integration.md (hunk 1 above)
Comment (xiaoyu-work) [reply docs/source/features/azureml_integration.md:4]: Updated.

File: docs/source/features/azureml_integration.md (hunk 2 above)
Comment (xiaoyu-work) [reply docs/source/features/azureml_integration.md:18]: Upated.

**Review by devang-ml — APPROVED** [review]

**Pull request merged**